File: tienda-backend-apirest/src/main/java/com/bolsadeideas/springboot/backend/apirest/models/entity/SaleEntity.java
package com.bolsadeideas.springboot.backend.apirest.models.entity;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "sale")
public class SaleEntity implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String date;
    private Integer productId;
    private String productName;
    private Integer amount;
    private Integer clientId;
    private String clientName;
    private Boolean pay;
    private Double price;
    private Long idShop;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public Integer getProductId() {
        return productId;
    }

    public void setProductId(Integer productId) {
        this.productId = productId;
    }

    public Integer getClientId() {
        return clientId;
    }

    public void setClientId(Integer clientId) {
        this.clientId = clientId;
    }

    public Boolean getPay() {
        return pay;
    }

    public void setPay(Boolean pay) {
        this.pay = pay;
    }


    public Integer getAmount() {
        return amount;
    }

    public void setAmount(Integer amount) {
        this.amount = amount;
    }

    public String getProductName() {
        return productName;
    }

    public void setProductName(String productName) {
        this.productName = productName;
    }

    public String getClientName() {
        return clientName;
    }

    public void setClientName(String clientName) {
        this.clientName = clientName;
    }

    public Double getPrice() {
        return price;
    }

    public void setPrice(Double price) {
        this.price = price;
    }

    public Long getIdShop() {
        return idShop;
    }

    public void setIdShop(Long idShop) {
        this.idShop = idShop;
    }
}

File: shop_app/lib/entitys/sale_entity.dart
class SaleEntity {
  int? id;
  String? date;
  int? productId;
  String? productName;
  int? amount;
  int? clientId;
  String? clientName;
  bool? pay;
  double? price;
  int? idShop;

  SaleEntity(
      {this.id,
        this.date,
        this.productId,
        this.productName,
        this.amount,
        this.clientId,
        this.clientName,
        this.pay,
        this.price,
        this.idShop
      });

  SaleEntity.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    date = json['date'];
    productId = json['productId'];
    productName = json['productName'];
    amount = json['amount'];
    clientId = json['clientId'];
    clientName = json['clientName'];
    pay = json['pay'];
    price = json['price'];
    idShop = json['idShop'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['date'] = this.date;
    data['productId'] = this.productId;
    data['productName'] = this.productName;
    data['amount'] = this.amount;
    data['clientId'] = this.clientId;
    data['clientName'] = this.clientName;
    data['pay'] = this.pay;
    data['price'] = this.price;
    data['idShop'] = this.idShop;
    return data;
  }
}